react/src/components/Footer.js

```diff
@@ -1,34 +1,61 @@
-import { Component } from 'react';
+import * as Sentry from '@sentry/react';
+import { Fragment, Component } from 'react';
 import { Link } from 'react-router-dom';
 import './footer.css';
 
 class Footer extends Component {
+  state = {
+    subscribed: false,
+  };
+
   async shouldComponentUpdate() {
     console.log('> Footer shouldComponentUpdate');
   }
 
   render() {
+    const handleSubscribeClick = () => {
+      this.setState({ subscribed: true });
+    };
+
     return (
       <footer id="footer">
         <div>
           <h2 className="h3 sentry-unmask">Sign up for plant tech news</h2>
-          <div class="formContainer">
-            <form>
-              <label htmlFor="email-subscribe" className="sentry-unmask">
-                Email
-              </label>
+          <Sentry.ErrorBoundary
+            onReset={() => {
+              this.setState({ subscribed: false });
+            }}
+            fallback={({ resetError, eventId }) => {
+              return (
+                <Fragment>
+                  <div>An error has occurred. Sentry Event ID: {eventId}</div>
+                  <button className="btn" onClick={resetError}>
+                    Reset Form
+                  </button>
+                </Fragment>
+              );
+            }}
+          >
+            <div className="formContainer">
+              <form>
+                <label htmlFor="email-subscribe" className="sentry-unmask">
+                  Email
+                </label>
+                <input
+                  type="email"
+                  name="email-subscribe"
+                  id="email-subscribe"
+                ></input>
+              </form>
               <input
-                type="email"
-                name="email-subscribe"
-                id="email-subscribe"
-              ></input>
-            </form>
-            <input
-              type="submit"
-              value="Subscribe"
-              className="sentry-unmask"
-            ></input>
-          </div>
+                type="submit"
+                value="Subscribe"
+                className="sentry-unmask"
+                onClick={handleSubscribeClick}
```

Comment:
Def not necessary, but consider adding an event handler for hitting the Enter key while on the input. The first time I tested out the new feature I hit Enter to submit the form by reflex, but the page refreshed with no problems instead of triggering the error.

[Example](https://thomasstep.com/blog/react-perform-an-action-when-enter-is-pressed)

Reply:
done with [d02442f](https://github.com/sentry-demos/empower/pull/314/commits/d02442fc0d3b204de3818f7598e32ab374424162)